core/src/commonMain/kotlin/com/ustadmobile/core/domain/filterusername/FilterUsernameUseCase.kt

```diff
@@ -0,0 +1,29 @@
+package com.ustadmobile.core.domain.filterusername
+
+import com.ustadmobile.core.domain.validateusername.ValidateUsernameUseCase
+
+/**
+ * Filters and normalizes username input:
+ * - Converts uppercase to lowercase
+ * - Filters invalid characters
+ * - Handles key event validation
+ */
+class FilterUsernameUseCase {
+    operator fun invoke(
+        username: String,
+        invalidCharReplacement: Char
```

Comment:
invalidCharReplacement should probably be changed to a string; because when it is being used by the viewmodel, the replacement for an invalid character is blank (empty string).